packages/@ourworldindata/core-table/src/Transforms.ts

```diff
@@ -340,6 +440,29 @@ export const applyTransforms = (
             )
             console.error(err)
         }
-    })
+    }
     return columnStore
 }
+
+export const extractDataSlugsFromTransform = (
+    transform: string
+): ColumnSlug[] => {
+    const { transformName, params = [] } =
+        extractTransformNameAndParams(transform) ?? {}
+    if (!transformName) return []
+    const dataParams = availableTransforms[transformName].params
+        .map((param, index) => ({ ...param, index }))
+        .filter(
+            (param) =>
+                param.type === TransformParamType.DataSlug ||
+                param.type === TransformParamType.ColumnSlug // might be a data slug
+        )
+    if (dataParams.length === 0) return []
+    const dataParamInds = dataParams.map((param) => param.index)
+    const dataSlugs = params.filter((_, index) => dataParamInds.includes(index))
```

Comment:
You can use `zip(availableTransforms[transformName].params, params)` maybe to make that all easier (lodash's `zip` works just like in Python), and to not have to deal with keeping track of indices.